config/routes.rb

```diff
@@ -1,20 +1,62 @@
 Rails.application.routes.draw do
   namespace :api, defaults: { format: :json } do
     namespace :v1 do
-      get "/merchants/random", to: "random_merchants#show"
-      get "/merchants/find", to: "finds_merchants#show"
-      get "/merchants/find_all", to: "finds_merchants#index"
+      namespace :merchants do
+        get "/random", to: "random_merchants#show"
+        get "/find", to: "finds_merchants#show"
+        get "/find_all", to: "finds_merchants#index"
+      end
       resources :merchants, only: [:index, :show] do
-        resources :items, only: [:index]
-      end
-      get "/customers/random", to: "random_customers#show"
-      get "/customers/find", to: "finds_customers#show"
-      get "/customers/find_all", to: "finds_customers#index"
-      resources :customers, only: [:index, :show]
-      get "/transactions/random", to: "random_transactions#show"
-      get "/transactions/find", to: "finds_transactions#show"
-      get "/transactions/find_all", to: "finds_transactions#index"
-      resources :transactions, only: [:index, :show]
+        resources :items, only: [:index], :controller => "merchants/merchant_items"
```

Comment:
This is a different syntax than what you are doing on lines 4 -8. Choose one over the other. Since you refactored to namespace controllers, you could move this to line 8 and do `get '/items', to: 'merchant_items#index'`